### PR TITLE
AFP-323 md metadata

### PR DESCRIPTION
**Pull request by Blasz**
This PR adds the ability to specify metadata for markdown pages.

Right now the metadata isn't used by brisk at all, it is only added to the site map data file with the intention of using it for nav related things.

Future tasks AFP-324, AFP-325 and AFP-326 involve utilising 'title', 'status' and 'keywords' fields. We may modify where the metadata goes or is uses as part of those tasks.

How it works:

1. At page generation time, we parse any markdown files related to pages that are generated (at this stage, project docs page, package docs page and package home (readme) page).
2. Metadata is specified as frontmatter, similar to https://www.gatsbyjs.org/docs/adding-markdown-pages/#including-frontmatter
3. We parse the markdown using remark + remark-frontmatter and then convert yaml to a JS object.
4. This is then added to the sitemap that is generated as part of page generation.

We're also adding the remark-frontmatter plugin to the next/mdx config to strip the frontmatter from the markdown file contents when they are loaded as JSX.

### CODE REVIEW

File: packages/website/dummy-data/simple-project/docs/guides/nifty-tricks/staying-at-netherfield.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+---
+worthIt: yes
```
Comment (dpisani-atl) [inline packages/website/dummy-data/simple-project/docs/guides/nifty-tricks/staying-at-netherfield.md]: +1

File: packages/website/dummy-data/simple-project/packages/mock-package1/README.md [hunk 2]
```diff
@@ -1,3 +1,8 @@
+---
+status: VFD
```
Comment (Noviny) [inline packages/website/dummy-data/simple-project/packages/mock-package1/README.md]: 💯

**Comment by Blasz** [comment]
> Some questions:
> Does parsing all of the markdown files during generation heavily impact performance?
Good question, I'll measure in tangerine & jira.

> Is write-pages the best place to extract the meta? IIRC brisk has a scanning phase (i.e get-package-info/get-docs-info) and a generate phase (write-pages), In my head the former is better for retrieving meta about the files.

Ah nice, yeah scanning phase is definitely a better place to put this, I'll put the logic there instead. Cheers.

**Comment by Blasz** [comment]
> Does parsing all of the markdown files during generation heavily impact performance?

I measured `build` in tangerine and the difference was roughly the same. I couldn't measure this in jira since `next build` takes 10+ mins.

I think we can measure this better if we expose the generation step as its own brisk command, e.g. `brisk generate`, and expose an option whether to parse markdown for meta or not. Probably better suited to its own task, I'll create a ticket for this.

**Comment by Noviny** [comment]
> brisk generate

We have a task to add this script (a lot of our testing loops are bad because of its absence) - probably add the option to start brisk without doing generate as well. Definitely its own task though.

**Pull request merged**